discovery-aws-api-async/src/main/scala/akka/discovery/awsapi/ecs/AsyncEcsSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,136 @@
+    val ecsClient = ECSAsyncClient.create()
+    implicit val ec: ExecutionContext = system.dispatcher
+    Future.firstCompletedOf(
+      Seq(
+        after(resolveTimeout, using = system.scheduler)(
+          Future.failed(new TimeoutException("Future timed out!"))
+        ),
+        resolveTasks(ecsClient, name).map(
+          tasks =>
+            Resolved(
+              serviceName = name,
+              addresses = for {
+                task <- tasks
+                container <- task.containers().asScala
+                networkInterface <- container.networkInterfaces().asScala
+              } yield
+                ResolvedTarget(
+                  host = networkInterface.privateIpv4Address(),
+                  port = None
+                )
+          )
+        )
+      )
+    )
+  }
+}
+
+object AsyncEcsSimpleServiceDiscovery {
+
+  // InetAddress.getLocalHost.getHostAddress throws an exception when running
+  // in awsvpc mode because the container name cannot be resolved.
+  // ECS provides a metadata file
+  // (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html)
+  // that we ought to be able to use instead to find our IP address, but the
+  // metadata file does not get set when running on Fargate. So this is our
+  // only option for determining what the canonical Akka and akka-management
+  // hostname values should be set to.
+  def getContainerAddress: Either[String, InetAddress] =
+    NetworkInterface.getNetworkInterfaces.asScala
+      .flatMap(_.getInetAddresses.asScala)
+      .filterNot(_.isLoopbackAddress)
+      .filter(_.isSiteLocalAddress)
+      .to[Seq] match {
+      case Seq(value) =>
+        Right(value)
+
+      case other =>
+        Left(s"Exactly one private address must be configured (found: $other).")
+    }
+
+  private def resolveTasks(ecsClient: ECSAsyncClient, serviceName: String)(
+      implicit ec: ExecutionContext): Future[Seq[Task]] =
+    for {
+      taskArns <- listTaskArns(ecsClient, serviceName)
+      task <- describeTasks(ecsClient, taskArns)
+    } yield task
+
+  private[this] def listTaskArns(ecsClient: ECSAsyncClient,
+                                 serviceName: String,
+                                 pageTaken: Option[String] = None,
+                                 accumulator: Seq[String] = Seq.empty)(
+      implicit ec: ExecutionContext): Future[Seq[String]] =
+    for {
+      listTasksResponse <- toScala(
+        ecsClient.listTasks(
+          ListTasksRequest
+            .builder()
+            .serviceName(serviceName)
+            .nextToken(pageTaken.orNull)
+            .desiredStatus(DesiredStatus.RUNNING)
+            .build()
+        )
+      )
+      accumulatedTasksArns = accumulator ++ listTasksResponse.taskArns().asScala
+      taskArns <- listTasksResponse.nextToken() match {
+        case null =>
+          Future.successful(accumulatedTasksArns)
+
+        case nextPageToken =>
```

Comment:
I've only tested with a maximum of two tasks, so this path hasn't been tested (and I don't see an obvious way to do so without spending launching 100+ on my personal account, which I'm not keen on doing) - so it's worth looking at this especially carefully.

Reply:
...thought about it for a few more seconds and realised that, IIRC, the SDK does support a maximum page size. So (_if_ I do recall correctly) it should be simple enough to do a manual test with a smaller number of tasks running to verify this.

Reply:
I've now covered this by (temporarily) adding `.maxResults(1)` to the request and bootstrapping a cluster of 5 task instances.

Reply:
Thanks for calling it out, we can make a separate ticket for testing this

Reply:
https://github.com/akka/akka-management/issues/145